src/state/gateway.py

```diff
@@ -13,21 +13,20 @@ class GatewayResourceDefinition:
     """Base class containing kubernetes resource definition.
 
     Attrs:
-        namespace: The gateway resource's namespace.
         gateway_name: The gateway resource's name
     """
 
-    namespace: str
+    # We're expecting more fields to be added.
     gateway_name: str
 
     @classmethod
     def from_charm(cls, charm: ops.CharmBase) -> "GatewayResourceDefinition":
         """Create a resource definition from charm instance.
 
         Args:
-            charm (ops.CharmBase): _description_
+            charm (ops.CharmBase): The charm instance.
```

Comment:
```suggestion
            charm: The charm instance.
```
Nitpick: We can remove the type hint in the docstring :D